core/src/appleTest/kotlin/com/datadog/kmp/DatadogExtTest.kt

```diff
@@ -0,0 +1,149 @@
+import com.datadog.kmp.core.configuration.ProxyType
+import com.datadog.kmp.core.configuration.UploadFrequency
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertTrue
+
+@Suppress("FunctionNaming")
+internal class DatadogExtTest {
+    // todo RUM-8122: support TrackerConsent and DatadogSite mappings after isEqual
+    //  method will be supported for them in ios code
+
+    @Test
+    fun `M return valid sdk value W SdkLogVerbosity_toSdkLogVerbosity`() {
+        LOG_VERBOSITY_MAP.entries
+            .associate { (key, value) -> value to key }
+            .assertAllKeysEqualToValuesWhen { it.toSdkLogVerbosity }
+    }
+
+    @Test
+    fun `M return valid native value W SdkLogVerbosity_native`() {
+        LOG_VERBOSITY_MAP
+            .assertAllKeysEqualToValuesWhen { it.native }
+    }
+
+    @Test
+    fun `M return valid native value W UploadFrequency_native`() {
+        mapOf(
+            UploadFrequency.FREQUENT to DDUploadFrequencyFrequent,
+            UploadFrequency.AVERAGE to DDUploadFrequencyAverage,
+            UploadFrequency.RARE to DDUploadFrequencyRare
+        )
+            .assertExhaustive(UploadFrequency.entries)
+            .assertAllKeysEqualToValuesWhen { it.native }
+    }
+
+    @Test
+    fun `M return valid native value W BatchProcessingLevel_native`() {
+        mapOf(
+            BatchProcessingLevel.HIGH to DDBatchProcessingLevelHigh,
+            BatchProcessingLevel.MEDIUM to DDBatchProcessingLevelMedium,
+            BatchProcessingLevel.LOW to DDBatchProcessingLevelLow
+        )
+            .assertExhaustive(BatchProcessingLevel.entries)
+            .assertAllKeysEqualToValuesWhen { it.native }
+    }
+
+    @Test
+    fun `M return valid native value W BatchSize_native`() {
+        mapOf(
+            BatchSize.LARGE to DDBatchSizeLarge,
+            BatchSize.MEDIUM to DDBatchSizeMedium,
+            BatchSize.SMALL to DDBatchSizeSmall
+        )
+            .assertExhaustive(BatchSize.entries)
+            .assertAllKeysEqualToValuesWhen { it.native }
+    }
+
+    @Test
+    fun `M return expected configuration W Configuration_native`() {
+        val expectedProxyConfig = mapOf(
+            "HTTPSEnable" to true,
+            "HTTPSProxy" to "hostname",
+            "HTTPSPort" to 8080u
+        )
+
+        val commonConfig = Configuration(
+            coreConfig = Configuration.Core(
+                batchSize = BatchSize.MEDIUM,
+                uploadFrequency = UploadFrequency.AVERAGE,
+                site = DatadogSite.US1_FED,
+                batchProcessingLevel = BatchProcessingLevel.MEDIUM,
+                trackCrashes = true,
+                proxyConfiguration = ProxyConfiguration(
+                    type = ProxyType.HTTP,
+                    hostname = expectedProxyConfig["HTTPSProxy"] as String,
+                    port = expectedProxyConfig["HTTPSPort"] as UInt
+                ),
+                backgroundTasksEnabled = true
```

Comment:
you can also use methods like `randomBoolean`, `randomEnum`, etc. to use random values